src/react-integration/hooks/useResourceNew.ts

```diff
@@ -21,46 +22,46 @@ function useOneResource<
 >(
   fetchShape: ReadShape<S, Params, Body>,
   params: Params | null,
-): CondNull<typeof params, Denormalized<S>> {
-  // maybePromise is undefined when data is stale or params is null
+): CondNull<typeof params, DenormalizedNullable<S>, Denormalized<S>> {
   const maybePromise = useRetrieve(fetchShape, params);
-  // resource is null when it is not in cache or params is null
-  const resource = useCacheNew(fetchShape, params);
-  const error = useError(fetchShape, params, resource);
+  const state = useContext(StateContext);
+  const [denormalized, ready] = useDenormalized(fetchShape, params, state);
+  const error = useError(fetchShape, params, ready);
 
-  if (!hasUsableData(resource, fetchShape) && maybePromise) throw maybePromise;
+  if (!hasUsableData(ready, fetchShape) && maybePromise) throw maybePromise;
```

Comment:
do you need to check the falsiness of maybePromise here

Reply:
confirmed that I do